### ISSUE TITLE
Powershell provisioner not interpolating template variables for google compute engine

### ISSUE THREAD

**Issue by jcoconnor** [hashicorp/packer#5878]
I'm trying to get familiar with the ```googlecompute``` builder but don't seem to be able get template variables interpolated correctly in a test powershell script.

In this case I'm referring to ```{{.Vars}}```, ```{{.Path}}``` in the powershell provider to simply print them on the output.
But these and the other selections are simply returning ```<no value>```, i.e.
```
"Write-Output 'Vars: {{.Vars}}'",
```
returns
```
2018/02/09 16:04:48 ui:     googlecompute: Vars: <no value>
```

I have also put in a few other examples, e.g. ```{{user `gce_project_id`}}``` & ```{{}}``` to see if these get translated correct, which they do (see log in gist referenced below):

On a separate note:
1. Is there a list of the valid template variables - I haven't located on ?
2. Specifically on the ```googlecompute``` builder - is it possible to get ```winrm_password``` returned [after it is generated](https://github.com/hashicorp/packer/blob/7f4112ba229309cfc0ebaa10ded2abdfaf1b22c8/builder/googlecompute/step_create_windows_password.go#L114), it does display in debug mode only (should I log this separately).

Sample json file used to reproduce: https://gist.github.com/0754068d8bf7bba664882f6c1cb17963
Associated Variables file: https://gist.github.com/41018109751d208f82fcaae42092e597

Log file: https://gist.github.com/jcoconnor/c983078fcb20512d794f793d5a8cf786
Packer command: ```PACKER_LOG=1 packer build -force -var 'headless=false' -var-file=vars.json -var 'winrm_timeout=12h' ../../common/google.base.json```

Packer Version: 1.1.1
Host: Mac OS - High Sierra
```Darwin Johns-MacBook-Pro-4.local 17.4.0 Darwin Kernel Version 17.4.0: Sun Dec 17 09:19:54 PST 2017; root:xnu-4570.41.2~1/RELEASE_X86_64 x86_64 i386 MacBookPro11,5 Darwin```

**Comment by SwampDragons** [comment]
Sorry I hadn't had a chance to read this closely and get you a response before now.

The tl;dr is that this is the expected behavior.

The {{.Vars}} template variable is only used in the [execute_command](https://www.packer.io/docs/provisioners/powershell.html#execute_command) option (and the elevated_execute_command option) to tell the provisioner where to place the variables within a command, and is not interpolated within the inline commands. 

If you want to access your environment variables inline, you just call each variable like you would if you were on the system... So in your example, the line `"write-output 'the value of Mumble is: $env:Mumble'", ` is how you'd do it. 

Unfortunately there is no list of the available template variables, but we're trying to improve our documentation to call out the variables that are available in the specific places they're available.

As to the question about the winrm_password from google_compute, we _just_ implemented (haven't even released yet) that functionality for AWS and we're planning to implement it for other builders; if this is something you'd like and don't see an issue for already, feel free to open one so you can track it.

I hope this answers your questions and apologize for taking so long -- In the future, if you think your issue is more of a usage question than a bug or a feature request, the [mailing list](https://groups.google.com/forum/#!forum/packer-tool) is generally a much faster option than waiting for someone to get to your issue.

**Comment by jcoconnor** [comment]
@SwampDragons (Megan) Thanks for the input - I was able to figure other workarounds in the end.

I will likely raise an issue with the ```winrm_password``` - again I have a workaround that allows us to the necessary provisioning, but it would be a lot neater to have access to the password.
